File: frontend/src/types/product.ts
/**
 * Product type definition
 * Represents a product in the bakery store
 * Compatible with backend API
 */
export interface Product {
  /** Unique identifier for the product */
  id: number | string;
  /** Product display name */
  name: string;
  /** Detailed product description */
  description: string;
  /** Price in Vietnamese Dong (VND) */
  price: number;
  /** Product category (e.g., "Kẹo", "Bánh", "Đồ uống") */
  category: string;
  /** Category ID from backend */
  categoryId?: number;
  /** Array of image URLs */
  images: string[];
  /** Whether the product is currently in stock */
  inStock: boolean;
  /** Whether product is active */
  isActive?: boolean;
  /** Average customer rating (0-5 stars) */
  rating?: number;
  /** Stock quantity available */
  stockQuantity?: number;
  /** Creation timestamp */
  createdAt?: string;
  /** Last update timestamp */
  updatedAt?: string;
}

/**
 * Convert product ID to number
 */
export function getProductId(product: Product): number {
  return typeof product.id === 'string' ? parseInt(product.id, 10) : product.id;
}

File: backend/Domain/Entities/Product.cs
namespace TiemBanhBeYeu.Api.Domain.Entities;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int StockQuantity { get; set; } = 0;
    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public bool InStock => StockQuantity > 0;
    public bool IsActive { get; set; } = true;
    public bool IsDeleted { get; set; } = false;
    public DateTime? DeletedAt { get; set; }
    public double? Rating { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
}
